crates/hir-ty/src/method_resolution.rs

```diff
@@ -952,7 +952,10 @@ fn iterate_method_candidates_with_autoref(
         )
     };
 
-    iterate_method_candidates_by_receiver(receiver_ty, first_adjustment.clone())?;
+    if !matches!(receiver_ty.value.kind(Interner), TyKind::Ref(Mutability::Mut, _, _)) {
```

Comment:
Looking at the rustc code, we should do this for immutable refs as well? https://github.com/rust-lang/rust/blob/f211da7101a3c91f0afc23436abbcd3bd1d40d2b/compiler/rustc_hir_typeck/src/method/probe.rs#L1182-L1189